src/generate/generic.rs

```diff
@@ -227,3 +227,62 @@ pub enum Variant<U, T> {
     ///Raw bits
     Res(U),
 }
+
+use Variant::*;
+impl<U, T> Variant<U, T> {
+    /// Check if the variant is expected
+    pub fn is_value(&self) -> bool {
+        match self {
+            Val(_) => true,
+            Res(_) => false,
+        }
+    }
+
+    /// Check if the variant is not expected
+    pub fn is_reserved(&self) -> bool {
+        match self {
+            Val(_) => false,
+            Res(_) => true,
+        }
+    }
+
+    /// Moves the value `v` out of the `Variant` if it is `Val(v)`.
+    ///
+    /// Panics if the self value equals `Res`
+    #[inline]
+    pub fn unwrap(self) -> T {
+        match self {
+            Val(v) => v,
+            Res(_) => panic!("Unexpected variant"),
+        }
+    }
+
+    /// Returns the contained value or a default
+    #[inline]
+    pub fn unwrap_or(self, def: T) -> T {
+        match self {
+            Val(v) => v,
+            Res(_) => def,
+        }
+    }
+
+    /// Returns the contained value or computes it from a closure
+    #[inline]
+    pub fn unwrap_or_else<F: FnOnce(U) -> T>(self, f: F) -> T {
+        match self {
+            Val(v) => v,
+            Res(u) => f(u),
+        }
+    }
+
+    /// Unwraps a result, yielding the content of an `Val`.
+    ///
+    /// Panics if the value is an `Res`, with a panic message including the
+    /// passed message, and the content of the `Res`.
+    pub fn expect(self, msg: &'static str) -> T {
```

Comment:
Do you really need `'static` bound here?

Reply:
It doesn’t compile without `'static`.

Reply:
Weird. The original expect doesn't have it.

Reply:
I think it's because original `expect` use formatted `panic!("{}", msg)`.
With 2 arguments.

https://doc.rust-lang.org/src/core/macros.rs.html#12